Multisig/Logic/Models/Safe.swift

```diff
@@ -159,6 +176,8 @@ extension Safe: Identifiable {
         if let addressString = deletedSafeAddress, let address = Address(addressString) {
             App.shared.notificationHandler.safeRemoved(address: address)
         }
+
+        updateCachedNames()
```

Comment:
interesting to see the effect on the device - if it makes these operations slower

Reply:
did not notice any slowdown, but could do it on a background thread

Reply:
ok